### DISCUSSION TITLE
Subscribe/events behavior

### DISCUSSION THREAD

**Discussion by Stibo** [vidstack/player#1352] [Q&A] [answered]
I'm very new to the subscribe logic for states, so I'm not sure if this behavior is normal:

1. I'm subscribing for "paused", "playing" and "ended" states and output the state to the console. Now I see that (i assume) after loading the video, "paused" is outputted to the console for every video, before anything happened. Is this normal?
2. Is it possible to subscribe to a "player ready" state or something like this? Like when the whole player is successfully loaded and prepared?
3. Is it bad to still use "player.addEventListener('play'..." etc. instead of subscribe?

**Comment by mihar-22** [comment, marked as answer]
1. Yes this is normal.

2. The `canPlay` property will notify you when the player is ready.

3. You can definitely listen to events, but when it comes to tracking state specifically it's not generally safe as you have to know when certain state changes between certain events. Prefer `subscribe` for tracking state changes, and events for actions (e.g, a `play` happened).

**Reply by Stibo** [reply]
Thanks!
